src/dataflow/src/source/mod.rs

```diff
@@ -1077,6 +1088,16 @@ where
         // Create activator for source
         let activator = scope.activator_for(&info.address[..]);
 
+        // This source will need to be activated when the durability frontier changes.
+        if let Some(wrapper) = timestamp_histories.as_mut() {
+            let durability_activator = scope.activator_for(&info.address[..]);
+            wrapper
+                .wrapper
+                .borrow_mut()
```

Comment:
This isn't wrong but so far all methods operating on `TimestampBindingBox` are mediated by corresponding methods on `TimestampBindingRc` which handle borrowing the RefCell. Should we add a `push_activator()` and keep `wrapper` private?

Reply:
I'm happy to make the change, but waiting on @ruchirK to get a read on whether this approach is tolerable at all. I think actually, the `durability_frontier` shouldn't be in this type, as it is entirely unrelated other than "also being shared by sources".